core/src/main/java/edu/wpi/grip/core/SwitchableSource.java

```diff
@@ -0,0 +1,23 @@
+package edu.wpi.grip.core;
+
+
+import java.io.IOException;
+
+/**
+ * A {@link Source} that can switch its value.
+ */
+public abstract class SwitchableSource extends Source {
+
+    /**
+     * Loads the next value that this source has into the socket.
+     * @throws IOException If the source fails to load the next value into the socket.
+     */
+    public abstract void nextValue() throws IOException;
```

Comment:
This pull request doesn't add that functionality. Its not what @bradamiller asked for. It may be what you want but that's not what is being added.